src/d3d9/d3d9_device.cpp

```diff
@@ -4059,20 +4040,13 @@ namespace dxvk {
       // No need to wait as its not in use.
       physSlice = pResource->GetMappedSlice(Subresource);
 
-      // We do not need to wait for the resource in the event the
-      // calling app promises not to overwrite data that is in use
-      // or is reading. Remember! This will only trigger for MANAGED resources
-      // that cannot get affected by GPU, therefore readonly is A-OK for NOT waiting.
-      const bool skipWait = (readOnly && managed) || scratch || (readOnly && systemmem && !wasWrittenByGPU);
```

Comment:
?

Reply:
Well readonly can write after all so we can't skip the wait here.

Reply:
Games are using it for writes but as I understood based on our conversation it was in a NO_OVERWRITE fashion which would make skipping waits okay.

The problem was I was aware of was that we weren't initially dirtying stuff.